File: iOS/UICollectionView/UICollectionView/UICollectionView/Category.swift
import UIKit

struct Category {
    let label: String
    let categoryIcon: UIImage?
}

extension Category {
    static func createDummy() -> [Category] {
        return [
            Category(label: "Groceries", categoryIcon: UIImage(named: "groceries")),
            Category(label: "Eating Out", categoryIcon: UIImage(named: "eating_out")),
            Category(label: "Household-supplies", categoryIcon: UIImage(named: "household_supplies")),
            Category(label: "Housing", categoryIcon: UIImage(named: "housing")),
            Category(label: "Insurance", categoryIcon: UIImage(named: "insurance")),
            Category(label: "Medical", categoryIcon: UIImage(named: "medical")),
            Category(label: "Transportation", categoryIcon: UIImage(named: "transportation")),
            Category(label: "Internet", categoryIcon: UIImage(named: "internet")),
            Category(label: "Cellphone", categoryIcon: UIImage(named: "cellphone")),
            Category(label: "Electricity", categoryIcon: UIImage(named: "electricity")),
            Category(label: "Gas", categoryIcon: UIImage(named: "gas")),
            Category(label: "Water", categoryIcon: UIImage(named: "water")),
            Category(label: "Beauty", categoryIcon: UIImage(named: "beauty")),
            Category(label: "Recreation", categoryIcon: UIImage(named: "recreation")),
            Category(label: "Savings", categoryIcon: UIImage(named: "savings")),
            Category(label: "Automotive", categoryIcon: UIImage(named: "automotive")),
            Category(label: "Clothing", categoryIcon: UIImage(named: "clothing")),
            Category(label: "Donation", categoryIcon: UIImage(named: "donation")),
            Category(label: "Education", categoryIcon: UIImage(named: "education")),
            Category(label: "Social", categoryIcon: UIImage(named: "social")),
            Category(label: "Special", categoryIcon: UIImage(named: "special")),
            Category(label: "Tax", categoryIcon: UIImage(named: "tax")),
            Category(label: "Television", categoryIcon: UIImage(named: "television")),
            Category(label: "Travel", categoryIcon: UIImage(named: "travel")),
            Category(label: "Other", categoryIcon: UIImage(named: "other"))
        ]
    }
}

File: Android/RecyclerView/app/src/main/java/tokyo/randx/portfolio/android/recyclerview/grid/Category.java
package tokyo.randx.portfolio.android.recyclerview.grid;

import java.util.ArrayList;
import java.util.List;

public class Category {
    String category;
    String displayName;

    Category(String category, String displayName) {
        this.category = category;
        this.displayName = displayName;
    }
    public static List<Category> createDummy() {
        List<Category> categories = new ArrayList<>();
        Category c1 = new Category("groceries", "Groceries");
        Category c2 = new Category("transportation", "Transportation");
        Category c4 = new Category("cellphone", "Cellphone");
        Category c5 = new Category("clothing", "Clothing");
        Category c6 = new Category("eating_out", "Eating_out");
        Category c7 = new Category("electricity", "Electricity");
        Category c8 = new Category("gas", "Gas");
        Category c9 = new Category("internet", "Internet");
        Category c10 = new Category("water", "Water");
        Category c11 = new Category("automotive", "Automotive");

        categories.add(c1);
        categories.add(c2);
        categories.add(c4);
        categories.add(c5);
        categories.add(c6);
        categories.add(c7);
        categories.add(c8);
        categories.add(c9);
        categories.add(c10);
        categories.add(c11);

        return categories;
    }
}
